File: modulemain/src/main/java/com/jason_sunyf/modulemain/MainActivity.java
package com.jason_sunyf.modulemain;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import com.jason_sunyf.core.base.BaseActivity;
import com.jason_sunyf.moduletraffic.MainTrafficActivity;
import com.jason_sunyf.moudlewhether.whether.MainWhetherActivity;

public class MainActivity extends BaseActivity implements View.OnClickListener {
    private Button btn1, btn2, btn3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mActionBarTitleTv.setText("模块化项目");
        setContentView(R.layout.activity_main);
        btn1 = (Button) findViewById(R.id.btn1);
        btn2 = (Button) findViewById(R.id.btn2);
        btn3 = (Button) findViewById(R.id.btn3);
        btn1.setOnClickListener(this);
        btn2.setOnClickListener(this);
        btn3.setOnClickListener(this);
    }

    @Override
    protected void initPresenter() {

    }

    @Override
    protected void initView() {

    }

    @Override
    protected void initDataAndEvent() {

    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.btn1) {
            startActivity(MainWhetherActivity.class);
        } else if (v.getId() == R.id.btn2) {
            startActivity(MainTrafficActivity.class);
        } else if (v.getId() == R.id.btn3) {
            startActivity(MainTrafficActivity.class);
        }
    }
}

File: modulemain/src/main/java/com/jason_sunyf/modulemain/MainActivity.kt
package com.jason_sunyf.modulemain

import android.os.Bundle
import android.view.View
import android.widget.Button
import com.jason_sunyf.core.base.BaseActivity
import com.jason_sunyf.core.base.BasePresenter
import com.jason_sunyf.moduletraffic.MainTrafficActivity
import com.jason_sunyf.moudlewhether.whether.MainWhetherActivity

class MainActivity : BaseActivity<BasePresenter<*>>(), View.OnClickListener {
    private var btn1: Button? = null
    private var btn2: Button? = null
    private var btn3: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mActionBarTitleTv.text = "模块化项目"
        setContentView(R.layout.activity_main)
        btn1 = findViewById<View>(R.id.btn1) as Button
        btn2 = findViewById<View>(R.id.btn2) as Button
        btn3 = findViewById<View>(R.id.btn3) as Button
        btn1!!.setOnClickListener(this)
        btn2!!.setOnClickListener(this)
        btn3!!.setOnClickListener(this)
    }

    override fun initPresenter() {

    }

    override fun initView() {

    }

    override fun initDataAndEvent() {

    }

    override fun onClick(v: View) {
        if (v.id == R.id.btn1) {
            startActivity(MainWhetherActivity::class.java)
        } else if (v.id == R.id.btn2) {
            startActivity(MainTrafficActivity::class.java)
        } else if (v.id == R.id.btn3) {
            startActivity(MainTrafficActivity::class.java)
        }
    }
}
